tutorials/t8_step5_element_data_c_interface.c

```diff
@@ -0,0 +1,355 @@
+              num_local_elements + num_ghost_elements);
+  /* Note: We will later need to associate this data with an sc_array in order to exchange the values for
+   *       the ghost elements, which we can do with sc_array_new_data (see t8_step5_exchange_ghost_data).
+   *       We could also have directly allocated the data here in an sc_array with
+   *       sc_array_new_count (sizeof (struct data_per_element), num_local_elements + num_ghost_elements);
+   */
+
+  /* Let us now fill the data with something.
+   * For this, we iterate through all trees and for each tree through all its elements, calling
+   * t8_forest_get_element_in_tree to get a pointer to the current element.
+   * This is the recommended and most performant way.
+   * An alternative is to iterate over the number of local elements and use
+   * t8_forest_get_element. However, this function needs to perform a binary search
+   * for the element and the tree it is in, while t8_forest_get_element_in_tree has a
+   * constant look up time. You should only use t8_forest_get_element if you do not know
+   * in which tree an element is.
+   */
+  {
+    t8_locidx_t         itree, num_local_trees;
+    t8_locidx_t         current_index;
+    t8_locidx_t         ielement, num_elements_in_tree;
+    t8_eclass_t         tree_class;
+    t8_eclass_scheme_c *eclass_scheme;
+    const t8_element_t *element;
+
+    /* Get the number of trees that have elements of this process. */
+    num_local_trees = t8_forest_get_num_local_trees (forest);
+    for (itree = 0, current_index = 0; itree < num_local_trees; ++itree) {
+      /* This loop iterates through all local trees in the forest. */
+      /* Each tree may have a different element class (quad/tri/hex/tet etc.) and therefore
+       * also a different way to interpret its elements. In order to be able to handle elements
+       * of a tree, we need to get its eclass_scheme, and in order to so we first get its eclass. */
+      tree_class = t8_forest_get_tree_class (forest, itree);
+      eclass_scheme = t8_forest_get_eclass_scheme (forest, tree_class);
+      /* Get the number of elements of this tree. */
+      num_elements_in_tree = t8_forest_get_tree_num_elements (forest, itree);
+      for (ielement = 0; ielement < num_elements_in_tree;
+           ++ielement, ++current_index) {
+        /* This loop iterates through all the local elements of the forest in the current tree. */
+
+        /* We can now write to the position current_index into our array in order to store
+         * data for this element. */
+        /* Since in this example we want to compute the data based on the element in question,
+         * we need to get a pointer to this element. */
+        element = t8_forest_get_element_in_tree (forest, itree, ielement);
+        /* We want to store the elements level and its volume as data. We compute these
+         * via the eclass_scheme and the forest_element interface. */
+    
```

Comment:
In the cxx-examples these lines are directly there where 'element_data' is filled. I think we should move these lines there too, such that one can take these two files next to each other and every line is the same except for the element calls. (See the following suggestions)
Otherwise, if you want to argue that this code is more clear (and it is), we should change it in the other tutorial to match this file.

Reply:
I won't quibble over that. Gonna commit your suggestion and move on. ;)